File: core/DailyTemperatures/DailyTemperatures.py
from typing import List


def dailyTemperatures(temperatures: List[int]) -> List[int]:
    length = len(temperatures)
    res = [0] * length

    stack = []
    for index, temperature in enumerate(temperatures):
        while stack and stack[-1][1] < temperature:
            res[stack[-1][0]] = index - stack[-1][0]
            stack.pop()
        stack.append((index, temperature))

    return res


if __name__ == '__main__':
    print(dailyTemperatures([73, 74, 75, 71, 69, 72, 76, 73]))

File: core/DailyTemperatures/DailyTemperatures.java
package com.dbc;

import java.util.Arrays;
import java.util.Stack;

public class DailyTemperatures {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Arrays.fill(res, 0);
        Stack<int[]> stack = new Stack<>();
        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && stack.peek()[1] < temperatures[i]) {
                res[stack.peek()[0]] = i - stack.peek()[0];
                stack.pop();
            }
            stack.add(new int[]{i, temperatures[i]});
        }

        return res;
    }
}
